File: 3016. Minimum Number of Pushes to Type Word II.go
func minimumPushes(word string) int {
    freq := make([]int, 26)
    for _, c := range word {
        freq[c-'a']++
    }
    sort.Slice(freq, func(i, j int) bool {
        return freq[i] > freq[j]
    })
    answer, count := 0, 0
    for _, v := range freq {
        answer += v * (count / 8 + 1)
        count++
    }
    return answer
}

File: java/3016. Minimum Number of Pushes to Type Word II.java
class Solution {
    public int minimumPushes(String word) {
        int[] freq = new int[26];
        for (char c : word.toCharArray()) freq[c-'a']++;
        Arrays.sort(freq);
        int answer = 0, count = 0;
        for (int i = 25; i >= 0; i--) {
            answer += freq[i]*(count++/8 + 1);
        }
        return answer;
    }
}